test/server.js

```diff
@@ -13,9 +13,61 @@ const config = {
     port: 5432,
     name: 'sensorthingstest',
     user: 'postgres',
-    pass: '12345678'
+    pass: '12345678',
+    hooks: {
+      'beforeValidate': (instance, options) => {
+        instance.properties = Object.assign({}, instance.properties, {
+          beforeValidate: 'beforeValidate'
+        });
+      },
+      'afterValidate': (instance, options) => {
+        instance.properties = Object.assign({}, instance.properties, {
+          afterValidate: 'afterValidate'
+        });
+      },
+      'beforeCreate': (instance, options) => {
+        instance.properties = Object.assign({}, instance.properties, {
+          beforeCreate: 'beforeCreate'
+        });
+      },
+      'afterCreate': (instance, options) => {
+        instance.properties = Object.assign({}, instance.properties, {
+          afterCreate: 'afterCreate'
+        });
+      },
+      'beforeDestroy': (instance, options) => {
+        instance.properties = Object.assign({}, instance.properties, {
+          beforeDestroy: 'beforeDestroy'
+        });
```

Comment:
I suspect setting an instance value on instance destruction hooks won't be useful for us. [Destroying an instance](http://docs.sequelizejs.com/en/latest/api/model/#destroyoptions-promiseinteger) does not return the instance, so you won't be able to check that the value was set on the instance. You need a different approach.